### ISSUE TITLE
Feature: Handle reboots automatically

### ISSUE THREAD

**Issue by bfred-it** [mholt/caddy#1451]
From what I understand, caddy is trying to be an easy-to-use good-defaults alternative to httpd and nginx.

It'd be great if it either took care keeping itself alive (e.g. with `caddy --setup-autorun`) or if you gave more instructions than:

> **How you choose to run Caddy is up to you.** Many users are satisfied with nohup caddy &. Others use screen. Users who need Caddy to come back up after reboots either do so in the script that caused the reboot, add a command to an init script, or configure a service with their OS.

**Comment by pbeckmann** [comment]
Hey,
at first its sound like a  great idea. But at a second...
I'm not very open to allow a single integrate it into the system. The way with nginx and httpd is a bit different, as most of the time, you'll install those applications using your package manager, which installs those scripts too.
And there are some documented ways to run caddy at https://github.com/mholt/caddy/tree/master/dist/init
In conclusion i would not integrate the run scripts into caddy. Cause most of the time they are not needed. And the needed scripts are transported in the release too.
Just my opinion :)
Greetings

**Comment by mholt** [comment]
Starting a process at boot is the job of the OS, not the process. As much as I hate how "easy" Linux is to use (or any other OS, for that matter), we're not in a good position to integrate directly with every OS and system configuration. I suggest following Peter's advice and check out the community-contributed init scripts and see if any of those can be adapted to your needs!
